### PR TITLE
buildozer: Remove unused loads with comments

### PR DESCRIPTION
**Pull request by vladmos**
Previously buildozer refused to remove an unused load if it had any comment, now it works only with a special "@unused" comment.

### CODE REVIEW

File: edit/fix.go [hunk 1]
```diff
@@ -396,6 +375,16 @@ func cleanUnusedLoads(f *build.File) bool {
 			all = append(all, load)
 		} else {
 			fixed = true
+			// If the load statement contains a post-comment, keep it
+			if len(load.Comment().After) == 0 {
+				continue
+			}
+
+			if len(all) == 0 {
```
Comment (laurentlb) [inline edit/fix.go:383]: Why do you test len(all)?

File: edit/fix.go [hunk 2]
```diff
@@ -396,6 +375,16 @@ func cleanUnusedLoads(f *build.File) bool {
 			all = append(all, load)
 		} else {
 			fixed = true
+			// If the load statement contains a post-comment, keep it
+			if len(load.Comment().After) == 0 {
```
Comment (laurentlb) [inline edit/fix.go:379]: Should we do the same for Comment().Before?

File: edit/fix.go (hunk 1 above)
Comment (vladmos) [reply edit/fix.go:383]: If `all` is not empty I can just attach the comments to the previous expression. My idea was that if there's an after-comment and we keep it, it belongs to the whole load-block and shouldn't be separated by an empty line: the following

    load(":a.bzl", "used")
    load(":b.bzl", "unused")
    # foo

becomes

    load(":a.bzl", "used")
    # foo

File: edit/fix.go (hunk 2 above)
Comment (vladmos) [reply edit/fix.go:379]: I'm not sure, there are indeed cases when a before-comment belongs only to one statement and to all of them. The fact that load statements are not separated by empty lines (unless they have comments) and that there's a (non-default) fix that orders them lexicographically makes it hard to determine which is the case:

    # Bar definitions:
    load(":bar.bzl", "foo")
    load(":foo.bzl", "bar")

vs

    # All definitions:
    load(":bar.bzl", "foo")
    load(":foo.bzl", "bar")

File: edit/fix.go (hunk 1 above)
Comment (laurentlb) [reply edit/fix.go:383]: I don't think the comment should be attached to another node. If we keep the comment, that means we consider it was incorrectly attached. So moving it to a block-comment might be better?

Comment (vladmos) [reply edit/fix.go]: Done.

File: edit/fix.go (hunk 2 above)
Comment (vladmos) [reply edit/fix.go]: Changed to keeping all comments (except suffix).

**Review by meisterT — APPROVED** [review]

**Pull request merged**